test/functional/spec-runner/context.js

```diff
@@ -30,8 +30,13 @@ class Thread {
 }
 
 class TestRunnerContext {
-  constructor() {
-    this.url = null;
+  constructor(opts) {
+    const d = { password: undefined, user: undefined, skipInit: false, useSessions: true };
+    opts = Object.assign({}, d, opts || {});
+    this.useSessions = opts.useSessions;
+    this.skipInit = opts.skipInit;
+    this.user = opts.user;
+    this.password = opts.password;
```

Comment:
I suppose useSessions is pretty self explanatory, but skipInit could use a bit of description, like what "init" exactly is being skipped. might as well put a little text saying when these props are useful.